components/camera/videosource/webcam.go

```diff
@@ -203,7 +205,29 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 	constraints := makeConstraints(attrs, debug, logger)
 
 	if attrs.Path != "" {
-		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		cam, err := tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		if err != nil {
+			return cam, err
+		}
+
+		utils2.PanicCapturingGo(func() {
+			ticker := time.NewTicker(time.Second / 2)
+			defer ticker.Stop()
+			defer utils2.TryClose(ctx, cam)
+
+			for {
+				select {
```

Comment:
in the for loop you should use the goutils.SelectContextOrWaitChan or SelectContextOrWait. The ticker isn't very necessary making the latter usage easier. The risk of the select as it is is that there's a 50/50 chance of each case happening if both case are eligible

Reply:
Oh fancy. Done!